zaza/openstack/charm_tests/neutron/tests.py

```diff
@@ -318,6 +318,253 @@ def test_security_checklist(self):
+            openstack_utils.get_neutron_session_client(cls.keystone_session))
+
+    def test_100_services(self):
+        """Verify the expected services are running."""
+        logging.debug('Checking system services on units...')
+
+        services = {
+            self.compute_unit.entity_id: ['nova-compute',
+                                          'neutron-plugin-openvswitch-agent'],
+            self.rabbitmq_unit.entity_id: ['rabbitmq-server'],
+            self.neutron_api_unit.entity_id: ['neutron-server'],
+        }
+
+        if self.current_os_release >= self.trusty_mitaka:
+            services[self.compute_unit.entity_id] = [
+                'nova-compute',
+                'neutron-openvswitch-agent'
+            ]
+
+        for unit_name, unit_services in services.items():
+            logging.info('Checking services on {}'.format(unit_name))
+            zaza.model.block_until_service_status(
+                unit_name=unit_name,
+                services=unit_services,
+                target_status='running',
+                pgrep_full=self.pgrep_full
+            )
+
+        logging.debug('OK')
+
+    def test_301_neutron_sriov_config(self):
+        """Verify data in the sriov agent config file."""
+        trusty_kilo = openstack_utils.get_os_release('trusty_kilo')
+        if self.current_os_release < trusty_kilo:
+            logging.debug('Skipping test, sriov agent not supported on < '
+                          'trusty/kilo')
+            return
+
+        zaza.model.set_application_config(
+            self.application_name,
+            {'sriov-device-mappings': 'physnet42:eth42',
+             'enable-sriov': 'True'})
+
+        zaza.model.wait_for_application_states()
+
+        # Config file affected by juju set config change
+        conf_file = '/etc/neutron/plugins/ml2/sriov_agent.ini'
+        expected = {
+            'sriov_nic': {
+                'physical_device_mappings': ['physnet42:eth42'],
+            },
+        }
+
+        zaza.model.block_until_oslo_config_entries_match(
+            self.application_name,
+            conf_file,
+            expected
+        )
+
+        # the CI environment does not expose an actual SR-IOV NIC to the
+        # functional tests. consequently the neutron-sriov agent will not
+        # run, and the charm will update its status as such. this will prevent
+        # the success of pause/resume test.
+        #
+        # disable sriov after validation of config file is complete.
+        logging.info('Disabling SR-IOV after verifying config file data...')
+
+        zaza.model.set_application_config(
+            self.application_name,
+            {'sriov-device-mappings': '',
+             'enable-sriov': 'False'})
+
+        logging.info('Waiting for config-changes to complete...')
+        zaza.model.wait_for_application_states()
+
+        logging.debug('OK')
+
+    def _check_ml2_setting_propagation(self, service, charm_key,
```

Comment:
It seems like this function could be generalized by passing in conf_file and then reused in more tests.